assets/js/leaves.js

```diff
@@ -0,0 +1,147 @@
+  let leftMid = x - leafSize.width / 2 + midXShift;
+  let rightMid = leftMid + leafWidth;
+
+  // if leaf is long enough, do two mid points
+  if (leafHeight > 50) {
+    // two mid points around thirds
+    let midY1 = leafHeight * 0.33 - randomNumBetween(7);
+    let midY2 = leafHeight * 0.66 - randomNumBetween(7);
+
+    // should we adjust the upper width?
+    if (cointoss()) {
+      // yes, adjust the first
+      // should we adjust the left or the right?
+      if (cointoss()) {
+        // left
+        // shrink width by adjustment on the leftside
+        let adjustment = randomNumBetween(7);
+        leftPoints.push([leftMid + adjustment, midY1]);
+      } else {
+        leftPoints.push([leftMid, midY1]);
+      }
+
+      if (cointoss()) {
+        // right
+        let adjustment = randomNumBetween(7);
+        rightPoints.unshift([rightMid - adjustment, midY1]);
+      } else {
+        rightPoints.unshift([rightMid, midY1]);
+      }
+    } else {
+      // push first width without adjusting
+      leftPoints.push([leftMid, midY1]);
+      rightPoints.unshift([rightMid, midY1]);
+
+      // adjust the second; should we adjust the left or the right?
+      let adjustment = randomNumBetween(5);
+      if (cointoss()) {
+        // left
+        // shrink width by adjustment on the leftside
+        leftPoints.push([leftMid + adjustment, midY2]);
+      } else {
+        leftPoints.push([leftMid, midY2]);
+      }
+      if (cointoss()) {
+        // right
+        rightPoints.unshift([rightMid - adjustment, midY2]);
+      } else {
+        rightPoints.unshift([rightMid, midY2]);
+      }
+    }
+  } else {
+    // for shorter leaves, only one set of midpoints
+    let midY = leafHeight / 2;
+    leftPoints.push([leftMid, midY]);
+    rightPoints.unshift([rightMid, midY]);
+  }
+
+  // tail of the leaf does not need to be centered
+  // start with midpoint of shifted middle, then shift slightly more
+  let tailX = leftMid + leafSize.width / 2 + plusOrMinus(leafSize.maxTipShift);
+  //let tailX = x + randomNumBetween(leafSize.maxTipShift, - leafSize.maxTipShift);
+  // may need some limits to limit shift based on midXShift
+
+  // how pointy is the tail? add points relative to the tail
+  // at 80% of the height, leaf should be on average 60% of the width; adjust randomly
+  let nearTailWidth = leafSize.width * 0.6 - randomNumBetween(7);
+  // let nearTailX = tailX - nearTailWidth; // randomNumBetween(nearTailWidth, - nearTailWidth);
+  // unadajusted tailx
+  let nearTailX = leftMid + leafSize.width / 2 - nearTailWidth / 2; // randomNumBetween(nearTailWidth, -
+  let nearTailY = leafHeight * 0.8;
+  if (cointoss()) {
+    leftPoints.push([nearTailX, nearTailY]);
+  } else {
+    // if (cointoss()) {
+    rightPoints.unshift([nearTailX + nearTailWidth, nearTailY]);
+    // }
+  }
+
+  // tail only needs to be added once
+  leftPoints.push([tailX, leafHeight]);
+
+  let leafPoints = leftPoints.concat(rightPoints);
```

Comment:
so the full array of points is left + right, arranged .. around the perimeter of the leaf? and the curve function draws through/between those lines?

Reply:
Yes, that's right. It seemed like an easier way to think about it than trying to do it sequentially, since the corresponding points on the sides need to be coordinated somewhat (at least the way I was thinking about it). Wondering about making a little object to hold the points, because I think it might be easier to manipulate and add the optional ones, but not sure it's needed.